File: CloudBridge-WD/src/main/java/de/r3pt1s/CloudBridge/network/protocol/packet/Packet.java
// This class was created by r3pt1s
package de.r3pt1s.CloudBridge.network.protocol.packet;

import de.r3pt1s.CloudBridge.network.protocol.ProtocolInfo;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;

abstract public class Packet extends ProtocolInfo {

    private ArrayList<Object> packetContent = new ArrayList<>();

    public void encode() {
        put(getId());
    }

    public void decode() {
        Object id = get();
    }

    public void put(Object value) {
        packetContent.add(value);
    }

    public Object get() {
        if (packetContent.size() > 0) {
            Object get = packetContent.get(0);
            packetContent.remove(0);
            Collection<Object> oldContent = new ArrayList<>(packetContent);
            packetContent.clear();
            packetContent.addAll(oldContent);
            return get;
        }
        return null;
    }

    public ArrayList<Object> getPacketContent() {
        return packetContent;
    }

    public void setPacketContent(ArrayList<Object> packetContent) {
        this.packetContent = packetContent;
    }

    public String getName() {
        return getClass().getSimpleName();
    }

    abstract public Integer getId();
}

File: CloudSystem/src/Cloud/network/protocol/packet/Packet.php
<?php

namespace Cloud\network\protocol\packet;

use Cloud\network\protocol\ProtocolInfo;

abstract class Packet extends ProtocolInfo {

    private array $packetContent = [];

    public function encode(): void {
        $this->put($this->getId());
    }

    public function decode(): void {
        $this->get();
    }

    public function put($value) {
        $this->packetContent[] = $value;
    }

    public function get(): mixed {
        if (count($this->packetContent) > 0) {
            $get = $this->packetContent[0];
            unset($this->packetContent[0]);
            $this->packetContent = array_values($this->packetContent);
            return $get;
        }
        return null;
    }

    public function getPacketContent(): array {
        return $this->packetContent;
    }

    public function setPacketContent(array $packetContent): void {
        $this->packetContent = $packetContent;
    }

    public function getName(): string {
        return (new \ReflectionClass($this))->getShortName();
    }

    abstract public function getId(): int;
}